object_storage_api/services/image.py

```diff
@@ -50,14 +52,19 @@ def create(self, image_metadata: ImagePostMetadataSchema, upload_file: UploadFil
 
         :param image_metadata: Metadata of the image to be created.
         :param upload_file: Upload file of the image to be created.
-        :return: Created image with an pre-signed upload URL.
+        :return: Created image with a pre-signed upload URL.
         :raises InvalidObjectIdError: If the image has any invalid ID's in it.
-        :raises InvalidFilenameExtension: If the image has a mismatched file extension.
+        :raises InvalidFilenameExtension: If the image has a mismatched file extension or the file extension is not
+            supported.
         """
 
-        expected_file_type = mimetypes.guess_type(upload_file.filename)[0]
+        file_extension = Path(upload_file.filename).suffix
+        if not file_extension or file_extension.lower() not in config.image.allowed_file_extensions:
+            raise FileTypeMismatchException(f"File extension `{upload_file.filename}` is not supported")
+
+        expected_file_type, _ = mimetypes.guess_type(upload_file.filename)
         if expected_file_type != upload_file.content_type:
-            raise InvalidFilenameExtension(
+            raise FileTypeMismatchException(
                 f"File extension `{upload_file.filename}` does not match content type `{upload_file.content_type}`"
```

Comment:
```suggestion
                f"File extension of '{upload_file.filename}' does not match content type `{upload_file.content_type}`"
```
I don't think this made sense before? E.g. `File extension `nvda_2024.4.2.exe` is not supported`